ros/angel_system_nodes/angel_system_nodes/intent_detector.py

```diff
@@ -0,0 +1,140 @@
+import json
+import time
+
+from cv_bridge import CvBridge
+import cv2
+import numpy as np
+import rclpy
+from rclpy.node import Node
+from std_msgs.msg import Header
+
+from angel_msgs.msg import InterpretedAudioUserIntent, Utterance
+
+# Please refer to labels defined in
+# https://docs.google.com/document/d/1uuvSL5de3LVM9c0tKpRKYazDxckffRHf7IAcabSw9UA .
+NEXT_STEP_KEYPHRASES = ['skip', 'next step']
+PREV_STEP_KEYPHRASES = ['prev step', 'last step', 'go back']
+OVERRIDE_KEYPHRASES = ['angel', 'angel system']
+
+# TODO(derekahmed): Please figure out how to keep this sync-ed with
+# config/angel_system_cmds/user_intent_to_sys_cmd_v1.yaml.
```

Comment:
Inline-thought: A possible method is checking set intersection with the keys in that command conversion config (maybe minus the "Other" label here for your failure catch).